src/sunmemory/sycl/sundials_sycl_memory.cpp

```diff
@@ -348,6 +336,6 @@ int SUNMemoryHelper_GetAllocStats_Sycl(SUNMemoryHelper helper,
     *bytes_allocated      = SUNHELPER_CONTENT(helper)->bytes_allocated_uvm;
     *bytes_high_watermark = SUNHELPER_CONTENT(helper)->bytes_high_watermark_uvm;
   }
-  else { return -1; }
-  return 0;
+  else { return SUN_ERR_EXT_FAIL; }
```

Comment:
You'd used `SUN_ERR_OUTOFRANGE` in the CUDA version and `SUN_ERR_CORRUPT` above in this file, but I think the most appropriate error code is
```suggestion
    return SUN_ERR_INCOMPATIBLE;
```